### DISCUSSION TITLE
cargo build failed while use the tokio 1.12.0

### DISCUSSION THREAD

**Discussion by Miss-you** [tokio-rs/tokio#4160] [Q&A]
env info:
```
mac
Darwin YOUSALI-MB1 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64
```

`cargo.toml `:
```
[package]
name = "fate-sync-lockstep-server"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
tokio = { version = "1.12.0", features = ["full"] }
...
```

error msg
```
Updating crates.io index
error: failed to select a version for the requirement `tokio = "^1.12.0"`
candidate versions found which didn't match: 1.11.0, 1.10.1, 1.10.0, ...
location searched: crates.io index
required by package `fate-sync-lockstep-server v0.1.0 (/Users/yousa/Documents/git/fate_sync/fate-sync-lockstep-server)`
```

**Comment by Noah-Kennedy** [comment]
Can you post the full `Cargo.toml`? This can sometimes happen due to annoying versioning conflicts, and it helps to know the rest of your dependencies when diagnosing these issues.

**Comment by Miss-you** [comment]
The `cargo.toml` is as follow. 
```
[package]
name = "fate-sync-lockstep-server"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
tokio = { version = "1.12.0", features = ["full"] }
tokio-util = { version = "0.6.8", features = ["full"] }
serde = { version = "1.0", features = ["derive"] }
serde_json = "1.0"
async-stream = "0.3.0"
atoi = "0.3.2"
bytes = "1"
structopt = "0.3.14"
tokio-stream = "0.1"

```

In addition, if I change it to `1.11.0`, `cargo build` will pass.

Best regards.

**Reply by Darksonn** [reply]
This `Cargo.toml` works fine for me. Please try running `cargo clean` and deleting your `Cargo.lock`.

**Reply by Miss-you** [reply]
> This `Cargo.toml` works fine for me. Please try running `cargo clean` and deleting your `Cargo.lock`.

It does not work. 
![image](https://user-images.githubusercontent.com/3816205/136640361-13dd9470-676f-4756-ba0b-1d0f8092eaba.png)

**Reply by Noah-Kennedy** [reply]
Ah, MacOS. I'll give it a go on my mac; I normally develop on Linux.

**Reply by Noah-Kennedy** [reply]
I was unable to reproduce.

**Reply by Noah-Kennedy** [reply]
What `rustc` version is this?

**Reply by Miss-you** [reply]
> What `rustc` version is this?

```
rustc --version
rustc 1.55.0
```

**Comment by Noah-Kennedy** [comment]
Cargo is not finding the latest tokio version in your crates.io index. I generally see this in offline builds, or builds where some sort of mirror of said index that is out-of-date is in use. Either way, this isn't an issue with tokio itself, but rather your local configuration.

**Reply by Miss-you** [reply]
> Cargo is not finding the latest tokio version in your crates.io index. I generally see this in offline builds, or builds where some sort of mirror of said index that is out-of-date is in use. Either way, this isn't an issue with tokio itself, but rather your local configuration.

I will use the `1.11.0` version first, thanks
